doc/source/whatsnew/v1.2.3.rst

```diff
@@ -16,6 +16,7 @@ Fixed regressions
 ~~~~~~~~~~~~~~~~~
 
 - Fixed regression in :meth:`~DataFrame.to_excel` raising ``KeyError`` when giving duplicate columns with ``columns`` attribute (:issue:`39695`)
+- Fixed regression in :class:`IntegerArray` unary ops propagating mask on assignment (:issue:`39943`)
```

Comment:
For future reference, I think we should generally say something like "nullable integer dtype" instead of "IntegerArray", as most users should never deal / know about IntegerArray

Reply:
Valid point, I can update this

Reply:
done. #40019. thanks @dsaxton